cmd/user-agent/src/pages/chapi/wallet/didcomm/mediator.js

```diff
@@ -77,31 +80,39 @@ export class AgentMediator {
     async isAlreadyConnected() {
         let res
         try {
-            res = await this.aries.mediator.getConnection()
+            res = await this.aries.mediator.getConnections()
         } catch (e) {
-            if (e.toString().includes("router not registered")) {
-                return false
-            }
-
             throw e
         }
 
-        return res.connectionID != ""
+        return res.connections && res.connections.length > 0
     }
 
     async createInvitation() {
         // creates invitation through the out-of-band protocol
-        let response = await this.aries.outofband.createInvitation({label: 'agent-label'})
+        let response = await this.aries.outofband.createInvitation({
+            label: 'agent-label',
+            router_connection_id: await getMediatorConnection(this.aries)
+        })
 
         return response.invitation
     }
 
     // TODO to be implemented
-    async requestDID(){
+    async requestDID() {
         console.warn('feature for requesting peer DID from router is not yet available')
     }
 }
 
+export async function getMediatorConnection(agent) {
+    let resp = await agent.mediator.getConnections()
+    if (resp.connections && resp.connections.length > 0) {
+        return resp.connections[0]
```

Comment:
This logic is being used both when creating OOB invitations and when accepting them as well.

Using the same router when creating or accepting OOB invitations attenuates the effectiveness of the pair-wise identifier (did:peer) shared in the did-exchange request (we now have a fixed router endpoint and keys in our out-going DIDs).

Please randomize this selection.

Reply:
For a followup: we should also consider asking the router for fresh routing keys. This would involve sending a new request that results in the same [`mediation-grant`](https://github.com/hyperledger/aries-rfcs/tree/master/features/0211-route-coordination#mediation-grant) message. The expectation is the routing keys and possibiy the endpoint would be different.